omegaconf/omegaconf.py

```diff
@@ -444,34 +461,14 @@ def _should_pass(special: str) -> bool:
                 )
             return ret
 
-        pass_parent = _should_pass("_parent_")
-        pass_root = _should_pass("_root_")
-
-        def resolver_wrapper(
-            config: BaseContainer,
-            parent: Container,
-            args: Tuple[Any, ...],
-            args_str: Tuple[str, ...],
-        ) -> Any:
-            if use_cache:
-                cache = OmegaConf.get_cache(config)[name]
-                try:
-                    return cache[args_str]
-                except KeyError:
-                    pass
-
-            # Call resolver.
-            kwargs = {}
-            if pass_parent:
-                kwargs["_parent_"] = parent
-            if pass_root:
-                kwargs["_root_"] = config
-
-            ret = resolver(*args, **kwargs)
-
-            if use_cache:
-                cache[args_str] = ret
-            return ret
+        resolver_wrapper = partial(
```

Comment:
Can you explain what using partial here buys?

Reply:
In order to check if the resolver was already registered to the same object, I need to access that object. There doesn't seem to be a "clean" way to achieve this with an inline function (maybe by messing with the closure, but I didn't dig too far as it looked like it would end up relying on CPython implementation details). On the other hand, a `partial` object offers a public API to access its data.

Anyway, since you suggested to drop the idea of re-registering the same function, I will revert this change.

Reply:
This has been reverted, new version is in 3554985